### ISSUE TITLE
Improve page titles to something better than "ORG_NAME | Tramline" 

### ISSUE THREAD

**Issue by pratul** [tramlinehq/tramline#580]
All the page titles (browser `<title>`s) are just "ORG_NAME | Tramline" right now. This means users must read the URL instead of relying on the page titles. 

It'll be better for usability if the titles reflected the purpose of the page. For e.g.

- 1.4.31 | Uber | Tramline 
- Settings | Uber Eats | Tramline 
- Production release | Uber Freight | Tramline

**Comment by Aaryanpal** [comment]
@kitallis will start with this.Please assign

in which of the below format i need to maintain the title?

> 1.4.31 | Uber | Tramline
> Settings | Uber Eats | Tramline
> Production release | Uber Freight | Tramline

**Comment by kitallis** [comment]
> @kitallis will start with this.Please assign
> 
> in which of the below format i need to maintain the title?
> 
> > 1.4.31 | Uber | Tramline
> > Settings | Uber Eats | Tramline
> > Production release | Uber Freight | Tramline

Hey @Aaryanpal, thanks for looking into this. 

So the examples above aren't necessarily formats, but options for different pages. I'll simplify it for you:

- Pages where app is available: `Page name | App Name | Tramline`
- Pages where app is not available `Page name | Org Name | Tramline`
- Live release page `Version Name (like 1.4.31) | App Name | Tramline` (there is only 1 live release page which is `/releases/slug-name/`

Hope that helps. Let me know if you have other questions.

**Comment by majjikishore007** [comment]
Hi, @kitallis  is this issue still open? if so, can I start working on this

**Comment by pratul** [comment]
Still open! Please raise a PR once you've taken a shot at it.

**Comment by majjikishore007** [comment]
Hi @kitallis  @pratul, May I know how the page title name should be framed. For example  

![image](https://github.com/user-attachments/assets/ab4e6b19-9653-470f-bfe1-2b91234191d9)
In the above screenshot, what should the page title be?  
Should it be the controller name + action or only the name of the controller?  
Or whenever the action == index, the page name would be == the controller name; otherwise, the page name == action.
Could you please simplify this? And here is one more case  
![image](https://github.com/user-attachments/assets/4e3984ba-4357-484f-937c-a920cb29a4ee)

**Comment by kitallis** [comment]
@majjikishore007 Thanks for thinking through this. I think this rule that you mentioned perhaps works the best:

```
 Or whenever the action == index, the page name would be == the controller name; otherwise, the page name == action.
```

But I think we need to do more than that. We should assign some kind of constant for every controller, that defines its title value; because the controller names don't necessarily map to what the users should see.

Here's a table for your reference on how and when to assign the action in front of the name.

| action  | action suffix in title? | user facing action? |
| ------------- | ------------- | ------------- |
| new  | yes  | yes |
| index  | no  | yes |
| show | no | yes |
| edit | yes | yes |
| create | no | no |
| destroy | no | no |
| update | no | no |

We'll of course ignore the actions that are not user-facing entirely.

So, to summarize, we'll do:

```
[PAGE TITLE CONSTANT] [ACTION SUFFIX IF APPLICABLE] | [APP or ORG] | Tramline

Releases | Ueno | Tramline
Train Edit | Ueno | Tramline
Apps | Ueno | Tramline
```

and, the previously mentioned,

```
[Version Name (like 1.4.31)] | [APP or ORG] | Tramline

1.3.31 | Ueno | Tramline
1.4.31 | Ueno | Tramline
```

Hope that helps.

**Comment by majjikishore007** [comment]
thanks @kitallis for giving more details. I have made the requested changes and made a PR. thanks
